mlflow/transformers.py

```diff
@@ -0,0 +1,793 @@
+import logging
+import pathlib
+
+from typing import Union, List, Optional, Dict, Any
+import yaml
+import warnings
+
+import mlflow
+from mlflow import pyfunc
+from mlflow.exceptions import MlflowException
+from mlflow.models import ModelInputExample, Model, infer_pip_requirements
+from mlflow.models.model import MLMODEL_FILE_NAME
+from mlflow.models.signature import ModelSignature
+from mlflow.models.utils import _save_example
+from mlflow.tracking._model_registry import DEFAULT_AWAIT_MAX_SLEEP_SECONDS
+from mlflow.utils.docstring_utils import format_docstring, LOG_MODEL_PARAM_DOCS
+from mlflow.utils.environment import (
+    _mlflow_conda_env,
+    _validate_env_arguments,
+    _CONDA_ENV_FILE_NAME,
+    _PYTHON_ENV_FILE_NAME,
+    _process_conda_env,
+    _process_pip_requirements,
+    _CONSTRAINTS_FILE_NAME,
+    _REQUIREMENTS_FILE_NAME,
+    _PythonEnv,
+)
+from mlflow.utils.file_utils import write_to
+from mlflow.utils.model_utils import (
+    _validate_and_copy_code_paths,
+    _validate_and_prepare_target_save_path,
+    _download_artifact_from_uri,
+    _get_flavor_configuration_from_uri,
+    _add_code_from_conf_to_system_path,
+)
+from mlflow.utils.requirements_utils import _get_pinned_requirement
+
+FLAVOR_NAME = "transformers"
+_PIPELINE_BINARY_KEY = "pipeline"
+_PIPELINE_BINARY_FILE_NAME = "pipeline.tr"
+_COMPONENTS_BINARY_KEY = "components"
+_TOKENIZER_KEY = "tokenizer"
+_FEATURE_EXTRACTOR_KEY = "feature_extractor"
+_IMAGE_PROCESSOR_KEY = "image_processor"
+_PROCESSOR_KEY = "processor"
+_TOKENIZER_TYPE_KEY = "tokenizer_type"
+_FEATURE_EXTRACTOR_TYPE_KEY = "feature_extractor_type"
+_IMAGE_PROCESSOR_TYPE_KEY = "image_processor_type"
+_PROCESSOR_TYPE_KEY = "processor_type"
+_CARD_TEXT_FILE_NAME = "model_card_text.txt"
+_CARD_DATA_FILE_NAME = "model_card_data.yaml"
+_TASK_KEY = "task"
+_LOGGED_TYPE_KEY = "logged_type"
+_INSTANCE_TYPE_KEY = "instance_type"
+_PIPELINE_MODEL_TYPE_KEY = "pipeline_model_type"
+_MODEL_PATH_OR_NAME_KEY = "source_model_name"
+
+_logger = logging.getLogger(__name__)
+
+
+def _model_engine_type(model) -> List[str]:
```

Comment:
Can we rename this function to `_model_libraries` or `_model_packages`?

Reply:
I like `_model_packages` :) changed!